### ISSUE TITLE
invisible cursor in term

### ISSUE THREAD

**Issue by kwon-young** [equalsraf/neovim-qt#145]
Since commit 11a6dcc886169992c306935bcd21b832ad2d3a68, the cursor in neovim terminal is not visible.
Example of neovim term in commit 11a6dcc886169992c306935bcd21b832ad2d3a68:
![spectacle tt3861](https://cloud.githubusercontent.com/assets/2647934/16172094/b52c4b6c-3580-11e6-8b5f-d3398a6ff4ca.png)
Example of neovim term in precedent commit 63a9f16f84c4888f5b4435f7a6e4d4f6bf180b93:
![spectacle et3861](https://cloud.githubusercontent.com/assets/2647934/16172103/fa505a3a-3580-11e6-9372-c848baad2571.png)

Cheers and thanks for your awesome work !

**Comment by equalsraf** [comment]
Interesting it only happens in terminal mode, also only seems to happen when the cursor is not on top of anything else.

Hypothesis: the terminal mode is explicitly drawing the cursor using the reverse attribute. But we draw our own cursor that inverts colors too.

e.g. this patch "fixes" the cursor in terminal mode by disabling the drawing of the cursor.

```diff
diff --git a/src/gui/shell.cpp b/src/gui/shell.cpp
index c445129..7cfe47d 100644
--- a/src/gui/shell.cpp
+++ b/src/gui/shell.cpp
@@ -534,7 +534,7 @@ void Shell::paintEvent(QPaintEvent *ev)
                }
                QPainter painter(this);
                painter.setCompositionMode(QPainter::RasterOp_SourceXorDestination);
-               painter.fillRect(cursorRect, m_cursor_color);
+//             painter.fillRect(cursorRect, m_cursor_color);
        }
 }
```

its not that the cursor is not there it just so happens the cursor is drawn by reversing the colors and the reverse color is the same as the background - for e.g. try testing with the colorscheme `blue` you should see the cursor then.

TLDR we need a better strategy to draw the cursor, or a way to avoid drawing the cursor in :term

**Comment by khalidchawtany** [comment]
@equalsraf If I apply this, I get the cursor back in terminal. However, I loose it in normal mode :(

**Comment by khalidchawtany** [comment]
I can get around invisible terminal cursor through setting the following:
```vim
  let g:terminal_color_7  = '#FBBC05'
```

The above is dark yellow and when the inverse is taken a dark blue cursor is displayed in the terminal.

**Comment by kwon-young** [comment]
Closing this issue, since I can't reproduce the bug.
